src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -637,6 +639,23 @@ String MergeTreeDataMergerMutator::getBestPartitionToOptimizeEntire(
         return {};
     }
 
+    if (max_total_size_to_merge && (*data_settings)[MergeTreeSetting::enable_max_bytes_limit_for_min_age_to_force_merge])
+    {
+        size_t total_sum_bytes = 0;
+
+        for (const auto & partition : partitions_info)
+            total_sum_bytes += partition.second.sum_bytes;
+
+        if (total_sum_bytes > max_total_size_to_merge)
+        {
+            LOG_TRACE(
+                log,
+                "Skip optimizing entire partition because of the total sum bytes of parts in partitions is greater than max_total_size_to_merge. Total sum bytes: {}, max_total_size_to_merge: {}",
+                total_sum_bytes, max_total_size_to_merge);
+            return {};
+        }
+    }
```

Comment:
you summed all partitions which doesn't make sense because we will merge only single partition
you need to modify the part below to something like:

```cpp
    const auto is_partition_invalid = (const auto & partition)
    {
        
        if (partition.num_parts == 1)
            return true;
            
        if (!max_total_size_to_merge || !(*data_settings)[MergeTreeSetting::enable_max_bytes_limit_for_min_age_to_force_merge])
            return false;
        
        return partition.sum_bytes > max_total_size_to_merge;
    };
    
    auto best_partition_it = std::max_element(
        partitions_info.begin(),
        partitions_info.end(),
        [&](const auto & e1, const auto & e2)
        {
            // If one partition cannot be used for some reason (e.g. it has only single part), always select the other partition.
            if (is_partition_invalid(e1.second))
                return true;
            if (is_partition_invalid(e2.second))
                return false;
            // If both partitions have more than one part, select the older partition.
            return e1.second.min_age < e2.second.min_age;
        });
```